RecoLocalCalo/HcalRecProducers/src/HcalESProducersGPUDefs.cc

```diff
@@ -0,0 +1,132 @@
+
+#include <iostream>
+
+using HcalRecoParamsGPUESProducer = HcalESProducerGPU<HcalRecoParamsRcd, HcalRecoParamsGPU, HcalRecoParams>;
+
+using HcalRecoParamsWithPulseShapesGPUESProducer =
+    HcalESProducerGPU<HcalRecoParamsRcd, HcalRecoParamsWithPulseShapesGPU, HcalRecoParams>;
+
+using HcalPedestalsGPUESProducer = HcalESProducerGPU<HcalPedestalsRcd, HcalPedestalsGPU, HcalPedestals>;
+
+using HcalGainsGPUESProducer = HcalESProducerGPU<HcalGainsRcd, HcalGainsGPU, HcalGains>;
+
+using HcalLUTCorrsGPUESProducer = HcalESProducerGPU<HcalLUTCorrsRcd, HcalLUTCorrsGPU, HcalLUTCorrs>;
+
+using HcalRespCorrsGPUESProducer = HcalESProducerGPU<HcalRespCorrsRcd, HcalRespCorrsGPU, HcalRespCorrs>;
+
+using HcalTimeCorrsGPUESProducer = HcalESProducerGPU<HcalTimeCorrsRcd, HcalTimeCorrsGPU, HcalTimeCorrs>;
+
+using HcalPedestalWidthsGPUESProducer =
+    HcalESProducerGPU<HcalPedestalWidthsRcd, HcalPedestalWidthsGPU, HcalPedestalWidths>;
+
+using HcalGainWidthsGPUESProducer = HcalESProducerGPU<HcalGainWidthsRcd, HcalGainWidthsGPU, HcalGainWidths>;
+
+using HcalQIECodersGPUESProducer = HcalESProducerGPU<HcalQIEDataRcd, HcalQIECodersGPU, HcalQIEData>;
+
+using HcalQIETypesGPUESProducer = HcalESProducerGPU<HcalQIETypesRcd, HcalQIETypesGPU, HcalQIETypes>;
+
+using HcalSiPMParametersGPUESProducer =
+    HcalESProducerGPU<HcalSiPMParametersRcd, HcalSiPMParametersGPU, HcalSiPMParameters>;
+
+using HcalSiPMCharacteristicsGPUESProducer =
+    HcalESProducerGPU<HcalSiPMCharacteristicsRcd, HcalSiPMCharacteristicsGPU, HcalSiPMCharacteristics>;
+
+DEFINE_FWK_EVENTSETUP_MODULE(HcalRecoParamsGPUESProducer);
+DEFINE_FWK_EVENTSETUP_MODULE(HcalRecoParamsWithPulseShapesGPUESProducer);
+DEFINE_FWK_EVENTSETUP_MODULE(HcalPedestalsGPUESProducer);
+DEFINE_FWK_EVENTSETUP_MODULE(HcalGainsGPUESProducer);
+DEFINE_FWK_EVENTSETUP_MODULE(HcalLUTCorrsGPUESProducer);
+DEFINE_FWK_EVENTSETUP_MODULE(HcalRespCorrsGPUESProducer);
+DEFINE_FWK_EVENTSETUP_MODULE(HcalTimeCorrsGPUESProducer);
+DEFINE_FWK_EVENTSETUP_MODULE(HcalPedestalWidthsGPUESProducer);
+DEFINE_FWK_EVENTSETUP_MODULE(HcalGainWidthsGPUESProducer);
+DEFINE_FWK_EVENTSETUP_MODULE(HcalQIECodersGPUESProducer);
+DEFINE_FWK_EVENTSETUP_MODULE(HcalQIETypesGPUESProducer);
+DEFINE_FWK_EVENTSETUP_MODULE(HcalSiPMParametersGPUESProducer);
+DEFINE_FWK_EVENTSETUP_MODULE(HcalSiPMCharacteristicsGPUESProducer);
+
+#include "FWCore/Framework/interface/eventsetuprecord_registration_macro.h"
+
+#include "RecoLocalCalo/HcalRecProducers/src/HcalCombinedRecordsGPU.h"
+
+#include "RecoLocalCalo/HcalRecAlgos/interface/HcalConvertedPedestalsGPU.h"
+#include "RecoLocalCalo/HcalRecAlgos/interface/HcalConvertedEffectivePedestalsGPU.h"
+#include "RecoLocalCalo/HcalRecAlgos/interface/HcalConvertedPedestalWidthsGPU.h"
+#include "RecoLocalCalo/HcalRecAlgos/interface/HcalConvertedEffectivePedestalWidthsGPU.h"
+
+EVENTSETUP_RECORD_REG(HcalConvertedPedestalsRcd);
+EVENTSETUP_RECORD_REG(HcalConvertedPedestalWidthsRcd);
```

Comment:
ok

Reply:
@makortel IIUC, in principle it would be enough to move the calls to `EVENTSETUP_RECORD_REG` to a separate package.

On the other hand it would be cleaner to move also the definition of the records in the same place, right ?

Reply:
Right, also the Record definitions should be moved into a non-plugin library.

Reply:
Could you check if 8d6456b6b999373532f3192d5ba95d35f016c2c8 addresses the issue ?

Reply:
> Could you check if [8d6456b](https://github.com/cms-patatrack/cmssw/commit/8d6456b6b999373532f3192d5ba95d35f016c2c8) addresses the issue ?

Looks good, thanks.